include/seqan3/io/stream/parse_condition.hpp

```diff
@@ -39,494 +39,104 @@
-// is_in_alphabet
-// ----------------------------------------------------------------------------
+//!\endcond
+inline detail::is_in_interval_type<interval_first, interval_last> constexpr is_in_interval{};
 
-/*!\brief Parse condition that checks if a given value is within the given alphabet `alphabet_t`.
+/*!\brief Checks whether a given letter is valid for the specified seqan3::alphabet_concept.
+ * \tparam alphabet_t The alphabet to check.
  * \ingroup stream
- * \implements seqan3::detail::parse_condition_concept
- * \tparam alphabet_t The alphabet type. Must satisfy the seqan3::alphabet_concept.
+ *
+ * \details
+ *
+ * This function like object returns true for all characters of the alphabet, false otherwise.
+ * The actual check being performed is whether assigning and then reading a letter results in the original input
+ * (but case is ignored).
+ *
+ * ### Example
+ *
+ * ```cpp
+ * is_in_alphabet<dna4>('C');  // returns true
+ *
+ * auto constexpr my_check = is_in_alphabet<dna4>;
+ * my_check('U');  // returns false, because it comes out as 'T'
+ * ```
  */
 template <alphabet_concept alphabet_t>
-struct is_in_alphabet : public detail::parse_condition<is_in_alphabet<alphabet_t>>
-{
-    //!\brief The message representing this condition.
-    static constexpr auto msg = constexpr_string{"is_in_alphabet<"} +
-                                constexpr_string{detail::get_display_name_v<alphabet_t>} +
-                                constexpr_string{">"};
-
-    //!\brief Inherit constructors from CRTP base class.
-    using detail::parse_condition<is_in_alphabet<alphabet_t>>::parse_condition;
-
-    /*!\brief Invokes the condition check for `val`.
-     * \tparam    char_t The type of the character to test. Must satisfy seqan3::char_adaptation_concept.
-     * \param[in] val    The value to test.
-     * \return `bool`; `true` if the condition is met, `false` otherwise.
-     *
-     * \details
-     *
-     * ### Complexity
-     *
-     * Constant
-     *
-     * ### Exception
-     *
-     * No-throw guarantee.
-     *
-     * ### Concurrency
-     *
-     * Thread-safe.
-     */
-    template <char_adaptation_concept char_t>
-    constexpr bool
-    operator()(char_t const val) const noexcept(std::is_same_v<char_t, char>)
-    {
-        if constexpr (!std::is_same_v<char_t, char>)
-        {  // Check if alphabet is able to represent the underlying char type of the tested alphabet.
-            if (static_cast<uint64_t>(val) >
-                static_cast<uint64_t>(std::numeric_limits<underlying_char_t<alphabet_t>>::max()))
-                return false;
-        }
-
-        return to_char(assign_char(alphabet_t{}, val)) == std::toupper(static_cast<uint8_t>(val));
-    }
-};
-
-// ----------------------------------------------------------------------------
-// is_char
-// ----------------------------------------------------------------------------
+inline detail::is_in_alphabet_type<alphabet_t> constexpr is_in_alphabet{};
```

Comment:
Please keep the const/constexpr consistent.
Meaning putting it in front of the type. (We really need automatic code refactorisation)

Reply:
I have recently started changing this, because we use east-const everywhere for `const` and it is very inconsistent to use east-const, but "west-constexpr".
I will update the code guide and at some point we should replace the other to be consistently east-const for variables, too.